src/services/tunables/tunables_service.cpp

```diff
@@ -61,29 +70,38 @@ namespace big
 						return;
 					}
 
+					m_tunables_backup = new std::uint64_t[m_num_tunables];
+					memcpy(m_tunables_backup, script_global(0x40000).as<void*>(), m_num_tunables * 8);
+
 					SCRIPT::SET_SCRIPT_WITH_NAME_HASH_AS_NO_LONGER_NEEDED("tuneables_processing"_J);
+					SCRIPT::SET_SCRIPT_WITH_NAME_HASH_AS_NO_LONGER_NEEDED("tunables_registration"_J);
 					m_script_started = true;
-
-					g_script_patcher_service->add_patch({"tuneables_processing"_J, "tuneables_processing1", "2E ? ? 55 ? ? 38 06", 0, std::vector<uint8_t>(17, 0x0), &m_script_started}); // bool tunables registration hack
-					if (auto program = gta_util::find_script_program("tuneables_processing"_J))
-						g_script_patcher_service->on_script_load(program);
 				}
 			}
 			else
 			{
 				if (SCRIPT::GET_NUMBER_OF_THREADS_RUNNING_THE_SCRIPT_WITH_THIS_HASH("tuneables_processing"_J) == 0)
 				{
+					for (int i = 0; i < m_num_tunables; i++)
+					{
+						auto value = *script_global(0x40000).at(i).as<int*>();
+						if (auto it = m_junk_values.find(value); it != m_junk_values.end())
+						{
+							m_tunables.emplace(it->second, 0x40000 + i);
+						}
+					}
+					memcpy(script_global(0x40000).as<void*>(), m_tunables_backup, m_num_tunables * 8);
+
 					if (m_tunables.size() == 0)
 					{
 						LOG(FATAL) << "Failed to cache tunables";
-						g_script_patcher_service->update();
 						return;
 					}
 
 					m_script_started = false;
 					m_initialized    = true;
 					LOG(INFO) << "Saving " << m_tunables.size() << " tunables to cache";
-					g_script_patcher_service->update();
+					delete[] m_tunables_backup;
```

Comment:
also set `m_tunables_backup` to `nullptr` as `delete` operator does nt do it.

Reply:
We should probably be using a unique_ptr instead of all this C nonsense memcpy crap, this isn't 2003 anymore. The unique pointer will release the memory when it falls out of scope.